### ISSUE TITLE
Gitea auth provider fails with 7.2.1

### ISSUE THREAD

**Issue by tobru** [oauth2-proxy/oauth2-proxy#1636]
## Expected Behavior

The Gitea auth provider should work as described in https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider#gitea-auth-provider.

## Current Behavior

Since version 7.2.1 Gitea auth provider fails:

```
[2022/05/01 19:00:37] [internal_util.go:74] token validation request failed: status 404 - 404 page not found
[2022/05/01 19:00:37] [internal_util.go:69] 404 GET https://git.example.com/api/v1 404 page not found
```

## Possible Solution

Downgrading to 7.2.0 makes login work again.

Most probably related to #1433.

## Steps to Reproduce (for bugs)

There is no step-by-step guide. Just use the Gitlab auth provider as documented.

## Context

Login with Gitea auth provider.

## Your Environment

Version used:
* Gitea 1.16.6
* Oauth2 Proxy 7.2.1

**Comment by mjpitz** [comment]
I just got bit by this and spent some time digging into it. It looks like the `OAUTH2_PROXY_VALIDATE_URL` is used in two different contexts. The first is by this `internal_util.go` file to ensure that the access token can be used with the API. The second is used to lookup the email address of the user associated with this token in the GitHub provider implementation:
https://github.com/oauth2-proxy/oauth2-proxy/blob/42475c28f7f235c08777a24027fe8ec170763482/providers/github.go#L425

As a result, you encounter 1 of 2 errors:
- if you provide the `VALIDATE_URL` as documented, then you encounter this internal_util error since it's attempting to call the `/api/v1` path, but that doesn't exist in Gitea (where it does in GitHub).
- if you provide  "VALIDATE_URL" that exists, then the `internal_util` check passes, but the lookup for the email address will fail (this is because we append the `/users/email` path the the validate url)

**Comment by JoelSpeed** [comment]
Does Gitea not behave in the same way as GitHub here, in that you can successfully access the API to validate the token works and then also use the API to fetch more data later?

What is the base Gitea API path normally?

**Comment by mjpitz** [comment]
The API is set up similarly to GitHub, but Gitea returns a 404 for `/api/v1` like GitHub does.

**Comment by JoelSpeed** [comment]
In that case I would guess we need to either carry some patch to make a separate Gitea provider, or, preferably, somehow make this work with separate profile and validate URLs.

Does Gitea have a `/users/email` path somewhere?

**Comment by mjpitz** [comment]
It does. All the other endpoints work fine, just not `/api/v1` on it's own. A simple option here would be to validate using the `/api/v1/users/email` endpoint rather than using `/api/v1` and `/api/v1/users/email` (it's not entirely clear to me why you need to check both, plus oidc more or less works this way by validating your token with the userinfo endpoint). This gets around the issue while allowing the provider to continue to work for GitHub while also supporting Gitea. I do agree that having a `gitea` provider would be much clearer from an end users perspective though.

**Comment by myers** [comment]
+1 for this bug to get fixed.  Just spent 3 hours trying to figure this out.

**Comment by tionis** [comment]
> This issue has been inactive for 60 days. If the issue is still relevant please comment to re-activate the issue. If no action is taken within 7 days, the issue will be marked closed.

This is still broken

**Comment by aellwein** [comment]
still broken...

**Comment by sfxworks** [comment]
It's adding an extra / for me
Sent to a 404 page with 
> https://my-gittea.server//login/oauth/authorize

**Comment by campbellhennessy** [comment]
Just ran into this issue as well, I had to downgrade to 7.2.0

**Comment by tionis** [comment]
Still not working and relevant

**Comment by clayauld** [comment]
I can confirm this is still broken on 7.4.0 as well. 

Downgrade to 7.2.0 resolved the issue. I will be using that version until this issue is fixed.

**Comment by bendlas** [comment]
I ran into this problem as well with 7.4.0

Instead of downgrading, I patched it with https://github.com/oauth2-proxy/oauth2-proxy/pull/1996

Everything seems to work.

**Comment by ajvpot** [comment]
This issue is still relevant.

**Comment by tuunit** [comment]
The next release is currently prepared 👍🏼 
I'll try to get the fix #2194 into the release.
